node/tests/integration/main.rs

```diff
@@ -418,7 +425,483 @@ async fn test_play_check() {
+    println!("----->>> Player 1 checked");
+    // Player 2 --> Check
+    check(&mut client, player2_account.id(), game_account_id, faucet_account_id, 2).await;
+    println!("----->>> Player 2 checked");
+    // Player 3 --> Check
+    check(&mut client, player3_account.id(), game_account_id, faucet_account_id, 3).await;
+    println!("----->>> Player 3 checked");
+
+    // check phase
+    let game_account = client.get_account(game_account_id).unwrap().0;
+    assert_eq!(
+        game_account.storage().get_item(CURRENT_PHASE_SLOT),
+        RpoDigest::new([Felt::from(4 as u8), Felt::ZERO, Felt::ZERO, Felt::ZERO])
+    );
+
+    println!("----->>> Showdown");
+}
+
+async fn bet(
+    client: &mut AzeClient,
+    player_account_id: AccountId,
+    game_account_id: AccountId,
+    faucet_account_id: AccountId,
+    player_bet: u8,
+    player_no: u8
+) {
+    fund_account(client, player_account_id, faucet_account_id).await;
+
+    let game_account = client.get_account(game_account_id).unwrap().0;
+    let game_account_storage = game_account.storage();
+
+    let last_raiser = game_account_storage.get_item(RAISER_INDEX_SLOT);
+    let last_phase_digest = game_account_storage.get_item(CURRENT_PHASE_SLOT);
+
+    let player_index: u8 = FIRST_PLAYER_INDEX + PLAYER_STATS_SLOTS * (player_no - 1);
+
+    let fungible_asset = FungibleAsset::new(faucet_account_id, BUY_IN_AMOUNT).unwrap();
+
+    let playbet_txn_data = PlayBetTransactionData::new(
+        Asset::Fungible(fungible_asset),
+        player_account_id,
+        game_account_id,
+        player_bet
+    );
+
+    let transaction_template = AzeTransactionTemplate::PlayBet(playbet_txn_data);
+    let txn_request = client.build_aze_play_bet_tx_request(transaction_template).unwrap();
+    execute_tx_and_sync(client, txn_request.clone()).await;
+
+    let note_id = txn_request.expected_output_notes()[0].id();
+    let note = client.get_input_note(note_id).unwrap();
+
+    let tx_template = TransactionTemplate::ConsumeNotes(game_account_id, vec![note.id()]);
+    let tx_request = client.build_transaction_request(tx_template).unwrap();
+    execute_tx_and_sync(client, tx_request).await;
+
+    // update the game account storage
+    let game_account = client.get_account(game_account_id).unwrap().0;
+    let game_account_storage = game_account.storage();
+    let player_index: u8 = (FIRST_PLAYER_INDEX + PLAYER_STATS_SLOTS * (player_no - 1));
+
+    // check next player index
+    assert_next_turn(&client, game_account_id, player_index, last_raiser, last_phase_digest).await;
+    // check highest bet
+    assert_eq!(
+        game_account_storage.get_item(HIGHEST_BET_SLOT),
+        RpoDigest::new([Felt::from(player_bet), Felt::ZERO, Felt::ZERO, Felt::ZERO])
+    );
+    // check player bet
+    assert_eq!(
+        game_account_storage.get_item((player_index + 3) as u8),
+        RpoDigest::new([Felt::from(player_bet), Felt::ZERO, Felt::ZERO, Felt::ZERO])
+    );
+}
+
+async fn check(

```

Comment:
Can we move these utility functions (check, bet, raise, call, fold) to some appropriate location (`tests/e2e/utils.rs`) since we are going to write few more e2e tests.